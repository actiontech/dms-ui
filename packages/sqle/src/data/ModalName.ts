export enum ModalName {
  // whitelist manage
  Add_Whitelist = 'ADD_WHITELIST',
  Update_Whitelist = 'UPDATE_WHITELIST',

  // rule template
  Clone_Rule_Template = 'CLONE_RULE_TEMPLATE',
  Export_Rule_Template = 'EXPORT_RULE_TEMPLATE',

  // audit plan
  Subscribe_Notice = 'SUBSCRIBE_NOTICE',
  Report_Record = 'REPORT_RECORD',

  // sqle management
  Assignment_Member_Single = 'Assignment_Member_Single',
  Assignment_Member_Batch = 'Assignment_Member_Batch',
  Change_Status_Single = 'Change_Status_Single',
  View_Audit_Result_Drawer = 'View_Audit_Result_Drawer',
  Change_SQL_Priority = 'Change_SQL_Priority',

  // IDE plugin audit
  View_Plugin_Audit_Result_Drawer = 'View_Plugin_Audit_Result_Drawer',

  //
  Create_Sql_Management_Exception = 'Create_Sql_Management_Exception',
  Update_Sql_Management_Exception = 'Update_Sql_Management_Exception',

  Pipeline_Configuration_Detail_Modal = 'Pipeline_Configuration_Detail_Modal',

  Version_Management_Release_Modal = 'Version_Management_Release_Modal',

  Version_Management_Execute_Modal = 'Version_Management_Execute_Modal',

  Version_Management_Associate_Workflow_Modal = 'Version_Management_Associate_Workflow_Modal',

  Version_Management_Offline_Execute_Modal = 'Version_Management_Offline_Execute_Modal'
}
