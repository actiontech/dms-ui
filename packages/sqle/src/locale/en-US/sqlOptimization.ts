// eslint-disable-next-line import/no-anonymous-default-export
export default {
  pageTitle: 'Sql optimization',
  ceTips:
    'When you need to get sql rewrite suggestions and optimize sql execution performance, you can use the sql optimization feature provided by the platform, and also get multiple optimization results such as sql rewriting, index optimization, and performance verification',
  table: {
    optimizationId: 'Optimization id',
    optimizationName: 'Name',
    instanceName: 'DB instance',
    dbType: 'Db type',
    performanceGain: 'Performance gain',
    createTime: 'Create time',
    status: 'Status',
    creator: 'Creator'
  },
  status: {
    finish: 'Optimization success',
    failed: 'Optimization failed',
    optimizing: 'Optimization in progress'
  },
  create: {
    linkButton: 'Create sql optimization',
    returnButton: 'Return to sql optimization list',
    successTips: 'Create sql optimization successfully',
    base: {
      title: 'Create sql optimization',
      name: 'Sql optimization name'
    },
    sqlInfo: {
      title: 'Input sql',
      dbType: 'Db type',
      instanceName: 'DB instance',
      instanceSchema: 'Database',
      uploadType: 'Select sql statement upload method',
      optimize: 'Optimize',
      tips: 'To improve the accuracy of the optimization results, the system will perform database analysis operations',
      format: 'Sql beautify',
      formatTips:
        'Currently, the database types that support sql beautify are {{supportType}}. if no data source is selected or the data source type selected is not yet supported, sql beautification may result in sql statement syntax errors.'
    }
  },
  overview: {
    sqlTable: {
      order: 'Order',
      sql: 'Sql text',
      syntaxError: 'Syntax error',
      recommendedIndex: 'Recommended index',
      hitIndex: 'Hit index',
      rewriteNumber: 'Rewrite optimization',
      performanceImprovement: 'Performance improvement',
      indexUsed: 'Index used',
      buttonText: 'Optimization details'
    },
    optimizationOverview: {
      title: 'Optimization overview',
      queryNumber: 'Parsed sql',
      rewriteNumber: 'Optimized sql',
      indexNumber: 'Optimized recommended index',
      performance: 'Performance improvement after optimization'
    },
    recommendedIndex: 'Recommended index',
    recommendedIndexTips:
      'Concentrate display of index recommendations generated based on each sql',
    sqlTableTitle: 'Sql statement list',
    indexTips: 'No index needs to be optimized',
    optimizingStatusTips:
      'Optimization in progress, please refresh to get the latest results',
    failedStatusTips: 'Optimization failed'
  },
  detail: {
    returnButton: 'Return to performance optimization overview',
    sqlRewrite: {
      title: 'Sql rewrite',
      originalSql: 'Original sql',
      optimizedSql: 'Rewritten sql'
    },
    triggeredRule: {
      title: 'Rule applied in this rewrite optimization'
    },
    recommenderIndex: {
      title: 'Recommended index'
    },
    performanceValidation: {
      title: 'Performance validation',
      performImprove: 'Performance improvement',
      performImproveDesc:
        'It is expected that the performance of this sql will improve after this optimization is implemented',
      beforePlan: 'Execution plan (before optimization)',
      afterPlan: 'Execution plan (after optimization)'
    }
  }
};
