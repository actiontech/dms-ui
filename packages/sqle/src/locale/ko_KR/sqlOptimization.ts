// eslint-disable-next-line import/no-anonymous-default-export
export default {
  pageTitle: 'SQL 최적화',
  ceTips:
    'SQL 재작성 제안을 얻고 SQL 실행 성능을 최적화해야 할 때, 플랫폼에서 제공하는 SQL 최적화 기능을 사용할 수 있으며, SQL 재작성, 인덱스 최적화, 성능 검증과 같은 여러 최적화 결과를 얻을 수 있습니다',
  table: {
    optimizationId: '최적화 ID',
    optimizationName: '이름',
    instanceName: 'DB 인스턴스',
    dbType: 'DB 유형',
    performanceGain: '성능 향상',
    createTime: '생성 시간',
    status: '상태',
    creator: '생성자'
  },
  status: {
    finish: '최적화 성공',
    failed: '최적화 실패',
    optimizing: '최적화 진행 중'
  },
  create: {
    linkButton: 'SQL 최적화 생성',
    returnButton: 'SQL 최적화 목록으로 돌아가기',
    successTips: 'SQL 최적화 생성 성공',
    base: {
      title: 'SQL 최적화 생성',
      name: 'SQL 최적화 이름'
    },
    sqlInfo: {
      title: 'SQL 입력',
      dbType: 'DB 유형',
      instanceName: 'DB 인스턴스',
      instanceSchema: '데이터베이스',
      uploadType: 'SQL 문 업로드 방법 선택',
      optimize: '최적화',
      tips: '최적화 결과의 정확도를 높이기 위해 시스템이 데이터베이스 분석 작업을 수행합니다',
      format: 'SQL 미화',
      formatTips:
        '현재 SQL 미화를 지원하는 데이터베이스 유형은 {{supportType}}입니다. 데이터 소스를 선택하지 않았거나 선택한 데이터 소스 유형이 아직 지원되지 않는 경우, SQL 미화로 인해 SQL 문 구문 오류가 발생할 수 있습니다.'
    }
  },
  overview: {
    sqlTable: {
      order: '순서',
      sql: 'SQL 텍스트',
      syntaxError: '구문 오류',
      recommendedIndex: '추천 인덱스',
      hitIndex: '히트 인덱스',
      rewriteNumber: '재작성 최적화',
      performanceImprovement: '성능 향상',
      indexUsed: '사용된 인덱스',
      buttonText: '최적화 세부 정보'
    },
    optimizationOverview: {
      title: '최적화 개요',
      queryNumber: '파싱된 SQL',
      rewriteNumber: '최적화된 SQL',
      indexNumber: '최적화된 추천 인덱스',
      performance: '최적화 후 성능 향상'
    },
    recommendedIndex: '추천 인덱스',
    recommendedIndexTips: '각 SQL을 기반으로 생성된 인덱스 추천 집중 표시',
    sqlTableTitle: 'SQL 문 목록',
    indexTips: '최적화가 필요한 인덱스가 없습니다',
    optimizingStatusTips:
      '최적화가 진행 중입니다. 최신 결과를 얻으려면 새로 고침하세요',
    failedStatusTips: '최적화 실패'
  },
  detail: {
    returnButton: '성능 최적화 개요로 돌아가기',
    sqlRewrite: {
      title: 'SQL 재작성',
      originalSql: '원본 SQL',
      optimizedSql: '재작성된 SQL'
    },
    triggeredRule: {
      title: '이 재작성 최적화에 적용된 규칙'
    },
    recommenderIndex: {
      title: '추천 인덱스'
    },
    performanceValidation: {
      title: '성능 검증',
      performImprove: '성능 향상',
      performImproveDesc:
        '이 최적화가 구현된 후 이 SQL의 성능이 향상될 것으로 예상됩니다',
      beforePlan: '실행 계획 (최적화 전)',
      afterPlan: '실행 계획 (최적화 후)'
    }
  }
};
