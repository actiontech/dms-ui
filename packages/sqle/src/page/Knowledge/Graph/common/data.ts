import { SigmaContainerProps } from '@react-sigma/core';
import { EdgeType, NodeType } from '../index.type';
import { Attributes } from 'graphology-types';

export const sigmaSettings: SigmaContainerProps<
  NodeType,
  EdgeType,
  Attributes
>['settings'] = {
  // 允许在无效容器中渲染 (默认值: false)
  allowInvalidContainer: true,
  defaultNodeType: 'circle',
  // 设置边的默认类型 (默认值: 'line')
  defaultEdgeType: 'arrow',

  // 相机缩放比例限制 (默认值: min: 0.1, max: 32)
  minCameraRatio: 0.2,
  maxCameraRatio: 5,

  // 节点渲染程序类 (默认值: {})
  nodeProgramClasses: {},

  // 渲染性能优化
  hideEdgesOnMove: true, // 移动时隐藏边 (默认值: false)
  hideLabelsOnMove: false, // 移动时隐藏标签 (默认值: false)
  renderLabels: true, // 是否渲染标签 (默认值: true)
  renderEdgeLabels: false, // 是否渲染边标签 (默认值: false)

  // 标签渲染优化
  labelDensity: 0.1, // 标签密度 (默认值: 0.03)
  labelGridCellSize: 100, // 标签网格大小 (默认值: 100)
  labelRenderedSizeThreshold: 8, // 标签大小阈值 (默认值: 6)

  // 默认样式
  defaultNodeColor: '#1a73e8', // 节点颜色 (默认值: '#999')
  defaultEdgeColor: '#ccc', // 边颜色 (默认值: '#ccc')

  // 交互体验优化
  zoomToSizeRatioFunction: (x: number) => x // 缩放比例函数 (默认值: Math.pow)
};
