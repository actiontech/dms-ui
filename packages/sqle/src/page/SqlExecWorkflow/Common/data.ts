export const SOURCE_WORKFLOW_PATH_KEY = 'sourceWorkflowId';
