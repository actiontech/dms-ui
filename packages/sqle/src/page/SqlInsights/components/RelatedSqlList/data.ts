import { GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum } from '@actiontech/shared/lib/api/sqle/service/SqlInsight/index.enum';
import { t } from '../../../../locale';

export const RelatedSqlFilterSourceOptions = [
  {
    label: t('sqlInsights.relatedSqlList.source.order'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workflow
  },
  {
    label: t('sqlInsights.relatedSqlList.source.sqlManage'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.sql_manage
  },
  {
    label: t('sqlInsights.relatedSqlList.source.workbench'),
    value: GetSqlPerformanceInsightsRelatedSQLFilterSourceEnum.workbench
  }
];
