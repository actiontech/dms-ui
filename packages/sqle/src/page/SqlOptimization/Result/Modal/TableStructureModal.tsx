import {
  SQLRenderer,
  BasicModal,
  EmptyBox,
  BasicEmpty,
  BasicInput
} from '@actiontech/shared';
import { useSelector, useDispatch } from 'react-redux';
import { IReduxState } from '../../../../store';
import { ModalName } from '../../../../data/ModalName';
import { updateSqlOptimizationModalStatus } from '../../../../store/sqlOptimization';
import { useTranslation } from 'react-i18next';
import { TableStructureModalWrapper } from './style';
import CopyButton from '../components/CopyButton';

const TableStructureModal: React.FC = () => {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const { modalStatus, currentTableData } = useSelector(
    (state: IReduxState) => ({
      modalStatus: state.sqlOptimization.modalStatus,
      currentTableData:
        state.sqlOptimization.tableStructureModal.currentTableData
    })
  );

  const open = !!modalStatus[ModalName.Sql_Optimization_Table_Structure_Modal];

  const handleCancel = () => {
    dispatch(
      updateSqlOptimizationModalStatus({
        modalName: ModalName.Sql_Optimization_Table_Structure_Modal,
        status: false
      })
    );
  };

  return (
    <BasicModal
      title={t('sqlOptimization.result.viewTableStructureAndOptimalIndex')}
      open={open}
      onCancel={handleCancel}
      footer={null}
      width={1000}
      centered
      maskClosable={false}
    >
      <TableStructureModalWrapper>
        <div className="section">
          <div className="section-header">
            <div className="section-title">
              {t('sqlOptimization.result.tableStructureForQuery')}
            </div>
            <EmptyBox if={!!currentTableData?.tableStructure}>
              <CopyButton content={currentTableData?.tableStructure} onlyIcon />
            </EmptyBox>
          </div>
          <div className="section-content">
            <EmptyBox
              if={!!currentTableData?.tableStructure}
              defaultNode={<BasicEmpty />}
            >
              <BasicInput.TextArea
                value={currentTableData?.tableStructure}
                readOnly
                rows={8}
                bordered={false}
              />
            </EmptyBox>
          </div>
        </div>

        <div className="section">
          <div className="section-header">
            <div className="section-title">
              {t('sqlOptimization.result.optimalIndexForQuery')}
            </div>
            <CopyButton
              content={currentTableData?.recommendedIndexes}
              onlyIcon
            />
          </div>
          <div className="section-content">
            <SQLRenderer
              wordWrap
              showLineNumbers
              sql={currentTableData?.recommendedIndexes}
            />
          </div>
        </div>
      </TableStructureModalWrapper>
    </BasicModal>
  );
};

export default TableStructureModal;
