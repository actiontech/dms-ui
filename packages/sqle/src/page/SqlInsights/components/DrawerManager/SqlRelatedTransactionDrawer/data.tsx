import { TransactionInfoLockTypeEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';
import { t } from '../../../../../locale';

export const TransactionInfoLockTypeEnumDict = {
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.exclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.shared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionShared'
  ),
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.intentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.sharedIntentionExclusive'
  ),
  [TransactionInfoLockTypeEnum.ROW_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.rowLock'
  ),
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.tableLock'
  ),
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: t(
    'sqlInsights.relatedSqlList.sqlRelatedTransaction.transactionInfo.lockTypeDict.metadataLock'
  )
};

export const TransactionInfoLockTypeEnumFlagDict = {
  [TransactionInfoLockTypeEnum.SHARED]: 'S',
  [TransactionInfoLockTypeEnum.INTENTION_SHARED]: 'IS',
  [TransactionInfoLockTypeEnum.INTENTION_EXCLUSIVE]: 'IX',
  [TransactionInfoLockTypeEnum.SHARED_INTENTION_EXCLUSIVE]: 'SIX',
  [TransactionInfoLockTypeEnum.ROW_LOCK]: 'RL',
  [TransactionInfoLockTypeEnum.TABLE_LOCK]: 'TL',
  [TransactionInfoLockTypeEnum.METADATA_LOCK]: 'ML',
  [TransactionInfoLockTypeEnum.EXCLUSIVE]: 'X'
};
