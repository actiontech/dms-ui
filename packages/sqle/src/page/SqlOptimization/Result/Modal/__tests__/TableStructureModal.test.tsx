import { superRender } from '@actiontech/shared/lib/testUtil/superRender';
import {
  getBySelector,
  mockUseCurrentProject,
  mockUseCurrentUser
} from '@actiontech/shared/lib/testUtil';
import { useDispatch, useSelector } from 'react-redux';
import { fireEvent, screen } from '@testing-library/react';
import { ModalName } from '../../../../../data/ModalName';
import TableStructureModal from '../TableStructureModal';
import { optimizationDetailMockData } from '@actiontech/shared/lib/testUtil/mockApi/sqle/sqlOptimization/data';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
  useDispatch: jest.fn()
}));

describe('sqle/SqlOptimization/Result/Modal/TableStructureModal', () => {
  const mockDispatch = jest.fn();

  const mockStateWithOpenModal = {
    sqlOptimization: {
      modalStatus: {
        [ModalName.Sql_Optimization_Table_Structure_Modal]: true
      },
      tableStructureModal: {
        currentTableData: {
          tableStructure: optimizationDetailMockData.metadata,
          recommendedIndexes:
            'CREATE INDEX idx_students_create_at ON students(create_at);'
        }
      }
    }
  };

  beforeEach(() => {
    mockUseCurrentProject();
    mockUseCurrentUser();
    (useDispatch as jest.Mock).mockImplementation(() => mockDispatch);
    jest.useFakeTimers();
    mockDispatch.mockClear();
    (useSelector as jest.Mock).mockImplementation((selector) =>
      selector(mockStateWithOpenModal)
    );
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
  });

  it('should render modal when open with data', () => {
    const { baseElement } = superRender(<TableStructureModal />);

    expect(screen.getByText('查看表结构及最佳索引')).toBeInTheDocument();

    expect(baseElement).toMatchSnapshot();
  });

  it('should handle modal close functionality', () => {
    superRender(<TableStructureModal />);

    fireEvent.click(getBySelector('.ant-modal-close-x'));

    expect(mockDispatch).toHaveBeenCalledWith({
      type: 'sqlOptimization/updateModalStatus',
      payload: {
        modalName: ModalName.Sql_Optimization_Table_Structure_Modal,
        status: false
      }
    });
  });
});
