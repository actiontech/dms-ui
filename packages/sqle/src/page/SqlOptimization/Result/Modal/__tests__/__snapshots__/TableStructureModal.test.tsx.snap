// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqle/SqlOptimization/Result/Modal/TableStructureModal should render modal when open with data 1`] = `
<body>
  <div />
  <div>
    <div
      class="ant-modal-root"
    >
      <div
        class="ant-modal-mask"
      />
      <div
        class="ant-modal-wrap ant-modal-centered"
        tabindex="-1"
      >
        <div
          aria-labelledby="test-id"
          aria-modal="true"
          class="ant-modal basic-modal-wrapper css-vvpsbb ant-zoom-appear ant-zoom-appear-prepare ant-zoom"
          role="dialog"
          style="width: 1000px;"
        >
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
          <div
            class="ant-modal-content"
          >
            <button
              aria-label="Close"
              class="ant-modal-close"
              type="button"
            >
              <span
                class="ant-modal-close-x"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  viewBox="0 0 14 14"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m7 6.175 2.887-2.887.825.825L7.825 7l2.887 2.888-.825.824L7 7.825l-2.888 2.887-.824-.824L6.175 7 3.288 4.113l.824-.825z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="ant-modal-header"
            >
              <div
                class="ant-modal-title"
                id="test-id"
              >
                查看表结构及最佳索引
              </div>
            </div>
            <div
              class="ant-modal-body"
            >
              <div
                class="css-1xpvlkz"
              >
                <div
                  class="section"
                >
                  <div
                    class="section-header"
                  >
                    <div
                      class="section-title"
                    >
                      该查询的表结构
                    </div>
                    <button
                      class="ant-btn ant-btn-default ant-btn-sm basic-button-wrapper css-geipcv"
                      title="复制"
                      type="button"
                    >
                      <span
                        class="ant-btn-icon"
                      >
                        <svg
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M4.667 4.667V2a.667.667 0 0 1 .666-.667H14a.667.667 0 0 1 .667.667v8.667a.667.667 0 0 1-.667.667h-2.667v2.662c0 .37-.3.67-.671.67H2.005a.67.67 0 0 1-.672-.67l.002-8.658c0-.37.3-.671.672-.671zm1.333 0h4.662c.37 0 .671.3.671.671V10h2V2.667H6z"
                            fill="#C3C6CD"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="section-content"
                  >
                    <textarea
                      class="ant-input ant-input-borderless basic-input-wrapper css-1pd1cd5"
                      placeholder="请输入{{name}}"
                      readonly=""
                      rows="8"
                    >
                      CREATE TABLE students (
    student_id INT AUTO_INCREMENT PRIMARY KEY,  -- 学生唯一标识
    student_name VARCHAR(100),                  -- 学生姓名
    phone_number VARCHAR(20),                   -- 学生电话号码
    create_at DATETIME,                         -- 学生账户创建时间
    age INT,                                    -- 学生年龄
);

CREATE TABLE student_classes (
    student_id INT PRIMARY KEY, -- 学生唯一标识
    class_id INT,               -- 班级唯一标识
);

CREATE TABLE enrollments (
    enrollment_id INT AUTO_INCREMENT PRIMARY KEY,  -- 选课记录ID
    student_id INT,                               -- 学生唯一标识
    course_id INT,                                -- 课程唯一标识
    enrollment_date DATETIME,                     -- 选课日期
);
                    </textarea>
                  </div>
                </div>
                <div
                  class="section"
                >
                  <div
                    class="section-header"
                  >
                    <div
                      class="section-title"
                    >
                      该查询的最佳索引
                    </div>
                    <button
                      class="ant-btn ant-btn-default ant-btn-sm basic-button-wrapper css-geipcv"
                      title="复制"
                      type="button"
                    >
                      <span
                        class="ant-btn-icon"
                      >
                        <svg
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M4.667 4.667V2a.667.667 0 0 1 .666-.667H14a.667.667 0 0 1 .667.667v8.667a.667.667 0 0 1-.667.667h-2.667v2.662c0 .37-.3.67-.671.67H2.005a.67.67 0 0 1-.672-.67l.002-8.658c0-.37.3-.671.672-.671zm1.333 0h4.662c.37 0 .671.3.671.671V10h2V2.667H6z"
                            fill="#C3C6CD"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <div
                    class="section-content"
                  >
                    <div
                      class="actiontech-sql-renderer-wrapper word-wrap css-q962ee"
                    >
                      <pre>
                        <code>
                          <div
                            class="code-line wrapped-line with-line-number"
                          >
                            <span
                              class="code-line-number"
                            >
                              1
                            </span>
                            <span
                              class="hljs-keyword"
                            >
                              CREATE
                            </span>
                             INDEX idx_students_create_at 
                            <span
                              class="hljs-keyword"
                            >
                              ON
                            </span>
                             students(create_at);
                          </div>
                        </code>
                      </pre>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-hidden="true"
            style="width: 0px; height: 0px; overflow: hidden; outline: none;"
            tabindex="0"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
